pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -300,21 +300,25 @@ func TestUpdatePod(t *testing.T) {
 			fakeClientset := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*item.prevPod}})
 			controller, podIndexer, _ := setupNewNoExecuteTaintManager(context.TODO(), fakeClientset)
 			controller.recorder = testutil.NewFakeRecorder()
-			go controller.Run(ctx)
 			controller.taintedNodes = item.taintedNodes
+			go controller.Run(ctx)
 
 			podIndexer.Add(item.prevPod)
 			controller.PodUpdated(nil, item.prevPod)
 
-			fakeClientset.ClearActions()
-			time.Sleep(timeForControllerToProgress)
+			if item.awaitForScheduledEviction {
+				nsName := types.NamespacedName{Namespace: item.prevPod.Namespace, Name: item.prevPod.Name}
```

Comment:
Can you run this tests 100 times to see if it's stable?

Reply:
Sure, pasting output of the last run for the executed command:
```sh
go test ./pkg/controller/nodelifecycle/scheduler/ -v -run "TestUpdatePod$" -count=100
```

```
=== RUN   TestUpdatePod
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled
I1114 14:48:34.460891 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.460968 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.461094 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node
I1114 14:48:34.471714 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.471812 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.471949 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_with_toleration
I1114 14:48:34.482521 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.482574 3040701 taint_manager.go:211] "Sending events to api server"
=== RUN   TestUpdatePod/removing_toleration
I1114 14:48:34.493840 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.493886 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.504168 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/lengthening_toleration_shouldn't_work
I1114 14:48:34.514886 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.515041 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:35.515644 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
--- PASS: TestUpdatePod (1.07s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_with_toleration (0.01s)
    --- PASS: TestUpdatePod/removing_toleration (0.02s)
    --- PASS: TestUpdatePod/lengthening_toleration_shouldn't_work (1.01s)
PASS
ok  	k8s.io/kubernetes/pkg/controller/nodelifecycle/scheduler	106.365s
=== RUN   TestUpdatePod
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled
I1114 14:48:34.460891 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.460968 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.461094 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node
I1114 14:48:34.471714 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.471812 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.471949 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_with_toleration
I1114 14:48:34.482521 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.482574 3040701 taint_manager.go:211] "Sending events to api server"
=== RUN   TestUpdatePod/removing_toleration
I1114 14:48:34.493840 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.493886 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.504168 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/lengthening_toleration_shouldn't_work
I1114 14:48:34.514886 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.515041 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:35.515644 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
--- PASS: TestUpdatePod (1.07s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_with_toleration (0.01s)
    --- PASS: TestUpdatePod/removing_toleration (0.02s)
    --- PASS: TestUpdatePod/lengthening_toleration_shouldn't_work (1.01s)
PASS
ok  	k8s.io/kubernetes/pkg/controller/nodelifecycle/scheduler	106.365s
=== RUN   TestUpdatePod
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled
I1114 14:48:34.460891 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.460968 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.461094 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node
I1114 14:48:34.471714 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.471812 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.471949 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/scheduling_onto_tainted_Node_with_toleration
I1114 14:48:34.482521 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.482574 3040701 taint_manager.go:211] "Sending events to api server"
=== RUN   TestUpdatePod/removing_toleration
I1114 14:48:34.493840 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.493886 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:34.504168 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
=== RUN   TestUpdatePod/lengthening_toleration_shouldn't_work
I1114 14:48:34.514886 3040701 taint_manager.go:206] "Starting NoExecuteTaintManager"
I1114 14:48:34.515041 3040701 taint_manager.go:211] "Sending events to api server"
I1114 14:48:35.515644 3040701 taint_manager.go:108] "NoExecuteTaintManager is deleting pod" pod="default/pod1"
--- PASS: TestUpdatePod (1.07s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_results_in_patch_and_delete_when_PodDisruptionConditions_enabled (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node (0.01s)
    --- PASS: TestUpdatePod/scheduling_onto_tainted_Node_with_toleration (0.01s)
    --- PASS: TestUpdatePod/removing_toleration (0.02s)
    --- PASS: TestUpdatePod/lengthening_toleration_shouldn't_work (1.01s)
PASS
ok  	k8s.io/kubernetes/pkg/controller/nodelifecycle/scheduler	106.365s
```

Reply:
FYI, in the last test which takes ~1s, the awaitForScheduledEviction section takes ~20ms. The 1s is spent in the `verifyPodActions` to await for processing of the eviction which results in observed delete aciton.